consensus/tendermint/committee/committee_set_test.go

```diff
@@ -0,0 +1,393 @@
+
+		gotCommittee := set.Committee()
+
+		if !reflect.DeepEqual(committeeMembers, gotCommittee) {
+			t.Fatalf("expected committee: %v and got: %v", committeeMembers, gotCommittee)
+		}
+	}
+
+	for _, size := range committeeSetSizes {
+		size := size
+		t.Run(fmt.Sprintf("get committee of %v member/s", size), func(t *testing.T) {
+			assertSetCommittee(t, size)
+		})
+	}
+}
+
+func TestSet_GetByIndex(t *testing.T) {
+	committeeMembers := createTestCommitteeMembers(t, 4, genRandUint64(4, maxSize))
+	sort.Sort(committeeMembers)
+	set, err := NewSet(copyMembers(committeeMembers), committeeMembers[0].Address)
+	assertNilError(t, err)
+
+	t.Run("can get member by index", func(t *testing.T) {
+		expectedMember := committeeMembers[1]
+		gotMember, err := set.GetByIndex(1)
+		assertNilError(t, err)
+
+		if !reflect.DeepEqual(expectedMember, gotMember) {
+			t.Fatalf("expected member: %v and got %v", expectedMember, gotMember)
+		}
+	})
+
+	t.Run("error on accessing member index not in committee", func(t *testing.T) {
+		_, err := set.GetByIndex(6)
+		assertError(t, consensus.ErrCommitteeMemberNotFound, err)
+	})
+}
+
+func TestSet_GetByAddress(t *testing.T) {
+	committeeMembers := createTestCommitteeMembers(t, 4, genRandUint64(4, maxSize))
+	sort.Sort(committeeMembers)
+	set, err := NewSet(copyMembers(committeeMembers), committeeMembers[0].Address)
+	assertNilError(t, err)
+
+	t.Run("can get member by Address", func(t *testing.T) {
+		expectedMember := committeeMembers[1]
+		index, gotMember, err := set.GetByAddress(expectedMember.Address)
+		assertNilError(t, err)
+
+		if index != 1 {
+			t.Fatalf("incorrect index of member expected: %v and got %v", 1, index)
+		}
+
+		if !reflect.DeepEqual(expectedMember, gotMember) {
+			t.Fatalf("expected member: %v and got %v", expectedMember, gotMember)
+		}
+	})
+
+	t.Run("error on accessing member address not in committee", func(t *testing.T) {
+		_, _, err := set.GetByAddress(common.HexToAddress("testaddress"))
+		assertError(t, consensus.ErrCommitteeMemberNotFound, err)
+	})
+}
+
+// TestSet_GetProposer tests the round robin selection of proposers by continuously calling GetProposer and checking
+// the resulting Committee is the same as ordered Committee passed to the set
+func TestSet_GetProposer(t *testing.T) {
+	numOfPasses := 10
+	setSizes := 100
+	for size := 1; size <= setSizes; size++ {
+		size := size
+		t.Run(fmt.Sprintf("check round robin for validator set size of %v", size), func(t *testing.T) {
+			committeeMembers := createTestCommitteeMembers(t, int64(size), genRandUint64(size, maxSize))
+			lastBlockProposer := committeeMembers[rand.Intn(size)].Address
+			sort.Sort(committeeMembers)
+
+			set, err := NewSet(copyMembers(committeeMembers), lastBlockProposer)
+			require.NoError(t, err)
+
+			firstCommitteeMemberAddr := committeeMembers[0].Address
+			startRound, endRound := 0, size
+			for i := 1; i <= numOfPasses; i++ {
+				if i > 1 {
+					startRound = endRound
+					endRound = size * i
+				}
```

Comment:
I think we could simplify this slightly by doing the following.
```suggestion
			var startRound, endRound int
			for i := 1; i <= numOfPasses; i++ {
				startRound = endRound
				endRound = size * i
```